core/src/test/java/org/elasticsearch/index/seqno/LocalCheckpointTrackerTests.java

```diff
@@ -253,7 +271,17 @@ public void testResetCheckpoint() {
         tracker.resetCheckpoint(localCheckpoint);
         assertThat(tracker.getCheckpoint(), equalTo((long) localCheckpoint));
         assertThat(tracker.getMaxSeqNo(), equalTo((long) maxSeqNo));
-        assertThat(tracker.processedSeqNo, empty());
+        assertThat(tracker.processedSeqNo, new BaseMatcher<LongObjectHashMap<FixedBitSet>>() {
```

Comment:
:(

Reply:
I despise `assertTrue`. Note that this gives a nicer error message if the assertion fails:

```
java.lang.AssertionError: 
Expected: empty
     but: was <[512=>org.apache.lucene.util.FixedBitSet@98761236]>
Expected :empty
     
Actual   :<[512=>org.apache.lucene.util.FixedBitSet@98761236]>
```

as opposed to

```
java.lang.AssertionError
	at __randomizedtesting.SeedInfo.seed([F9E3DE31F5D979AE:C4C54170CE328934]:0)
	at org.junit.Assert.fail(Assert.java:86)
	at org.junit.Assert.assertTrue(Assert.java:41)
	at org.junit.Assert.assertTrue(Assert.java:52)
	at org.elasticsearch.index.seqno.LocalCheckpointTrackerTests.testResetCheckpoint(LocalCheckpointTrackerTests.java:276)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
```